utils/paramfetch/src/lib.rs

```diff
@@ -48,18 +48,25 @@ struct ParameterData {
 }
 
 #[inline]
-fn param_dir() -> String {
-    std::env::var(DIR_ENV).unwrap_or_else(|_| PARAM_DIR.to_owned())
+fn param_dir(data_dir: &Path) -> PathBuf {
+    std::env::var(PathBuf::from(DIR_ENV))
+        .map(PathBuf::from)
+        .unwrap_or_else(|_| data_dir.join(PARAM_DIR))
+}
+
+pub fn set_proofs_parameter_cache_dir_env(data_dir: PathBuf) {
+    std::env::set_var(DIR_ENV, data_dir.join(PARAM_DIR));
 }
```

Comment:
If `FIL_PROOFS_PARAMETER_CACHE` is already set then `param_dir()` would return one thing and `set_proofs_parameter_cache_dir_env` would set the environment variable to something completely different. That's not what we want. If `FIL_PROOFS_PARAMETER_CACHE` is present, it should be honored.

```suggestion
pub fn set_proofs_parameter_cache_dir_env(data_dir: PathBuf) {
    std::env::set_var(param_dir(data_dir));
}
```